mevislab.github.io/content/tutorials.md

```diff
@@ -0,0 +1,225 @@
+| `.dcm` | DCM part of the imported DICOM file, see [Importing DICOM Data](./tutorials/basicmechanisms/dataimport#DICOMImport "Importing DICOM Data"). |
+
+### Types of Modules {#Module_Types}
+
+Within the concept of MeVisLab the basic entities we are working with
+a graphical representations of modules having their specific functions
+for image processing, image visualization, and image interaction.
+
+The three basic module types (ML, Open Inventor and Macro Module) are
+distinguished by their colors:
+
+| <div style="width:230px">Type</div> | <div style="width:430px">Look</div> | Characteristics |
+| --- | --- | --- |
+| ML Module (blue) | ![ML Module](/images/MLMModuleML.png "ML Module") | Page-based, demand-driven processing of voxels |
+| Open Inventor Module (green) | ![Open Inventor Module](/images/MLMModuleSo.png "Open Inventor Module") | Visual scene graphs (3D); naming convention: all modules starting with *So* (for **s**cene **o**bject) |
+| Macro Module (brown) | ![Macro Module](/images/MLMModuleMacro.png "Macro Module") | Combination of other module types, allowing implementing hierarchies and scripted interaction |
+
+If a module is invalid, it is displayed in bright red. This might happen if the module itself is not available on your system.
+
+
+| <div style="width:330px">Appearance</div> | Explanation |
+| --- | --- |
+| ![Invalid Module](/images/MLMModuleStateInvalid.png "Invalid Module") | Invalid module |
+  ![Macro State Invalid](/images/MLMModuleStateMacroInvalidModule.png "Macro State Invalid") | Macro containing an invalid module |
+
+The number of warning and error messages that are being printed to the
+debug console are listed at the upper right corner of the module. Once
+the debug console is cleared, the warning and error indicators at the
+module are also cleared. If the module produces information messages,
+their number is printed in gray at this position. This enables a network
+or module developer to find the modules in a network that produce
+messages quickly.
+
+### Module Connectors {#Module_Connectors}
+Most modules have connectors which are displayed on the module. These represent the inputs (bottom) and outputs (top) of modules.
+
+In MeVisLab, three types of connectors are defined.
+
+| <div style="width:330px">Look</div> | <div style="width:130px">Appearance</div> | Definition |
+| --- | --- | --- |
+| ![Triangle](/images/MLMConnectorTriangle.png "Triangle - ML Image") | triangle | ML images |
+| ![Circle](/images/MLMConnectorHalfCircle.png "Circle - Inventor Scene") | half-circle | Inventor scene |
+| ![Square](/images/MLMConnectorSquare.png "Square - Base Object") | square | Base objects: pointers to data structures |
+
+By connecting these connectors and therefore establishing a so-called data connection, image data, or Open Inventor information is transported from one module to one or more others.
```

Comment:
Connecting two of these connectors establishs a so-called data connection. Image data or Open Inventor state information is transported between modules via these connections.